### PR TITLE
[SPARK-45265][SQL][BUILD][FOLLOWUP] Add `-Xss64m` for Maven testing of the `sql/hive` module

### PR DESCRIPTION
**Pull request by LuciferYang**
### What changes were proposed in this pull request?
This pr add `-Xss64m` for Maven testing of the `sql/hive` module

### Why are the changes needed?
Use test options consistent with sbt and fix the mavren daily test
- https://github.com/apache/spark/actions/runs/11879474992/job/33101447051
- https://github.com/apache/spark/actions/runs/11879098676/job/33100585352
- https://github.com/apache/spark/actions/runs/11881833077/job/33106633785

```
*** RUN ABORTED ***
An exception or error caused a run to abort: MetaException(message:Exception rolling back to savepoint rollback_5974738954110) 
  org.apache.hadoop.hive.ql.metadata.HiveException: MetaException(message:Exception rolling back to savepoint rollback_5974738954110)
  at org.apache.hadoop.hive.ql.metadata.Hive.addPartitions(Hive.java:3592)
  at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
  at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
  at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
  at java.base/java.lang.reflect.Method.invoke(Method.java:569)
  at org.apache.spark.sql.hive.client.Shim_v4_0.createPartitions(HiveShim.scala:1437)
  at org.apache.spark.sql.hive.client.HiveClientImpl.$anonfun$createPartitions$1(HiveClientImpl.scala:663)
  at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.scala:18)
  at org.apache.spark.sql.hive.client.HiveClientImpl.$anonfun$withHiveState$1(HiveClientImpl.scala:302)
  at org.apache.spark.sql.hive.client.HiveClientImpl.liftedTree1$1(HiveClientImpl.scala:233)
  ...
  Cause: org.apache.hadoop.hive.metastore.api.MetaException: Exception rolling back to savepoint rollback_5974738954110
  at org.apache.hadoop.hive.metastore.ObjectStore$GetHelper.run(ObjectStore.java:4508)
  at org.apache.hadoop.hive.metastore.ObjectStore.addPartitionsInternal(ObjectStore.java:2811)
  at org.apache.hadoop.hive.metastore.ObjectStore.addPartitions(ObjectStore.java:2729)
  at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
  at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
  at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
  at java.base/java.lang.reflect.Method.invoke(Method.java:569)
  at org.apache.hadoop.hive.metastore.RawStoreProxy.invoke(RawStoreProxy.java:97)
  at jdk.proxy25/jdk.proxy25.$Proxy252.addPartitions(Unknown Source)
  at org.apache.hadoop.hive.metastore.HMSHandler.add_partitions_core(HMSHandler.java:4304)
  ...
```

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
- https://github.com/LuciferYang/spark/runs/33114760431

### Was this patch authored or co-authored using generative AI tooling?
No

### CODE REVIEW

**Comment by LuciferYang** [comment]
cc @yaooqinn FYI

**Review by yaooqinn — APPROVED** [review]

File: .github/workflows/build_and_test.yml [hunk 1]
```diff
@@ -1198,3 +1198,9 @@ jobs:
           cd ui-test
           npm install --save-dev
           node --experimental-vm-modules node_modules/.bin/jest
+
+  maven-test:
```
Comment (LuciferYang) [inline .github/workflows/build_and_test.yml:1202]: will revert this after test

File: sql/hive/pom.xml [hunk 2]
```diff
@@ -237,7 +237,7 @@
         <artifactId>scalatest-maven-plugin</artifactId>
         <configuration>
           <!-- Specially disable assertions since some Hive tests fail them -->
-          <argLine>-da -Xmx4g -XX:ReservedCodeCacheSize=${CodeCacheSize} ${extraJavaTestArgs}</argLine>
+          <argLine>-da -Xmx4g -Xss64m -XX:ReservedCodeCacheSize=${CodeCacheSize} ${extraJavaTestArgs}</argLine>
```
Comment (yaooqinn) [inline sql/hive/pom.xml:240]: Can we comment both here and in SparkBuild that we should sync these arguments?

Comment (LuciferYang) [reply sql/hive/pom.xml:240]: will add later

**Comment by LuciferYang** [comment]
image

`sql/hive` others passed

**Review by yaooqinn — APPROVED** [review]

**Review by zhengruifeng — APPROVED** [review]

**Comment by yaooqinn** [comment]
Merged to master, thank you @LuciferYang @zhengruifeng

**Review by dongjoon-hyun — COMMENTED** [review]
+1, LGTM.

**Pull request closed without merging**